### PR TITLE
Allow BlockClock to scale and maintain 1/3 of the available window real estate

### PR DESCRIPTION
**Pull request by jarolrod**
Follow up can allow to customize this scaling in settings.

#### small window size

| master | pr |
| ------ | -- |
| master-1 |  Screen Shot 2023-04-20 at 2 55 23 PM |

#### large window size

| master | pr |
| ------ | -- |
| master-2 |  Screen Shot 2023-04-20 at 2 55 28 PM |

[Windows](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/unsecure_win_gui.zip?branch=pull/306)
[Intel macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/unsecure_mac_gui.zip?branch=pull/306)
[Apple Silicon macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/unsecure_mac_arm64_gui.zip?branch=pull/306)
[ARM64 Android](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/unsecure_android_apk.zip?branch=pull/306)

### CODE REVIEW

File: src/qml/components/BlockClock.qml [hunk 1]
```diff
@@ -28,7 +30,9 @@ Item {
 
     BlockClockDial {
         id: dial
-        anchors.fill: parent
+        width: Math.min((root.parentWidth / 3), (root.parentHeight / 3))
```
Comment (jarolrod) [inline src/qml/components/BlockClock.qml:33]: Allowing for different scale factors is as easy as changing the number 3 here

File: src/qml/components/BlockClock.qml [hunk 2]
```diff
@@ -12,9 +12,11 @@ import "../controls"
 
 Item {
     id: root
+    property real parentWidth: 600
```
Comment (johnny9) [inline src/qml/components/BlockClock.qml:15]: I don't think these properties are needed. You can reference the parent with root.parent.width and root.parent.height.

File: src/qml/components/BlockClock.qml [hunk 3]
```diff
@@ -95,13 +99,24 @@ Item {
...
+        indicatorDimensions: dial.width * (3/200)
+        indicatorSpacing: dial.width / 40
         paused: root.paused
     }
 
+    NetworkIndicator {
+        id: networkIndicator
+        property int topMargin: dial.width * (3/20)
+        anchors.top: dial.bottom
+        textSize: dial.width * (3/40)
```
Comment (johnny9) [inline src/qml/components/BlockClock.qml:115]: I understand why you did it here to match the original design, but I think making the size of this component depend on the size of the dial creates the wierd logic with having to know more about the parents layout than it should.

**Review by johnny9 — COMMENTED** [review]
I think the scaling logic for the dial looks great. I'm not sure about the network indicator scaling and having the two scale off of the their parent's parent width/height. It breaks the encapsulation.

**Comment by jarolrod** [comment]
Updated from [87b9f59 ](https://github.com/bitcoin-core/gui-qml/commit/87b9f59753e105a5cd73e3b3a62d727680a8ece0)to [71a1d99](https://github.com/bitcoin-core/gui-qml/commit/71a1d99a0c6c0fa3db8a884bdf04a238b40d704c), [compare](https://github.com/bitcoin-core/gui-qml/compare/87b9f59753e105a5cd73e3b3a62d727680a8ece0..71a1d99a0c6c0fa3db8a884bdf04a238b40d704c)

Changes:
- block clock takes up half the space
- Network Indicator doesn't scale, stays fixed size

File: src/qml/components/BlockClock.qml (hunk 2 above)
Comment (jarolrod) [reply src/qml/components/BlockClock.qml:15]: I think its better for clarity, and this in a small way documents what the proportions are based on by default - coming from the design file

**Comment by jarolrod** [comment]
Updated from [71a1d99 ](https://github.com/bitcoin-core/gui-qml/commit/71a1d99a0c6c0fa3db8a884bdf04a238b40d704c)to [d138d1c](https://github.com/bitcoin-core/gui-qml/commit/d138d1c06873dab390ca89d80e22cb337ddecbf4), [compare](https://github.com/bitcoin-core/gui-qml/compare/71a1d99a0c6c0fa3db8a884bdf04a238b40d704c..d138d1c06873dab390ca89d80e22cb337ddecbf4)

Changes: block clock is now 5/12 in size by default, this is halfway between 1/2 and 1/3.

**Review by johnny9 — APPROVED** [review]
ACK d138d1c06873dab390ca89d80e22cb337ddecbf4

**Comment by thebagmaster** [comment]
ACK [cf6e327](https://github.com/bitcoin-core/gui-qml/commit/cf6e3275564eff6254baa92d9de4cf394c95fb71)

**Comment by jarolrod** [comment]
Closing this in favor of keeping everything in https://github.com/bitcoin-core/gui-qml/pull/311

**Pull request closed without merging**